sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadJwtBearerTokenAuthenticationConverter.java

```diff
@@ -1,102 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-package com.azure.spring.cloud.autoconfigure.aad;
-
-import com.azure.spring.cloud.autoconfigure.aad.implementation.constants.AadJwtClaimNames;
-import com.azure.spring.cloud.autoconfigure.aad.implementation.constants.AuthorityPrefix;
-import com.azure.spring.cloud.autoconfigure.aad.implementation.jwt.AadJwtGrantedAuthoritiesConverter;
-import com.azure.spring.cloud.autoconfigure.aad.implementation.oauth2.AadOAuth2AuthenticatedPrincipal;
-import com.azure.spring.cloud.autoconfigure.aad.properties.AadResourceServerProperties;
-import org.springframework.core.convert.converter.Converter;
-import org.springframework.security.authentication.AbstractAuthenticationToken;
-import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.oauth2.core.OAuth2AccessToken;
-import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal;
-import org.springframework.security.oauth2.jwt.Jwt;
-import org.springframework.security.oauth2.server.resource.authentication.BearerTokenAuthentication;
-import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
-import org.springframework.util.Assert;
-
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-
-
-/**
- * A {@link Converter} that takes a {@link Jwt} and converts it into a {@link BearerTokenAuthentication}.
- *
- * @deprecated use the default converter {@link JwtAuthenticationConverter} instead in {@link AadResourceServerWebSecurityConfigurerAdapter}.
- */
-@Deprecated
```

Comment:
@chenrujun please help confirm.

Reply:
@chenrujun @saragluna I would like to know the alternative solution for `AADJwtBearerTokenAuthenticationConverter`, I am using this to validate the user role.

Reply:
We changed to use the default one `JwtAuthenticationConverter`, the default claim-to-authority map is:
"scp" -> "SCOPE_", 
"roles" -> "APPROLE_"
you can custom via the below properties:
```properties
spring.cloud.azure.active-directory.resource-server.claim-to-authority-prefix-map.roles=ROLE_
```

Or customize the jwt granted authorities converter via below code:

```java
@Bean
@ConditionalOnBean({ AadResourceServerProperties.class})
SecurityFilterChain defaultAadResourceServerFilterChain(HttpSecurity http) throws Exception {
	Converter<Jwt, Collection<GrantedAuthority>> xxxJwtGrantedAuthoritiesConverter = null;
	http.apply(aadResourceServer()
			.jwtGrantedAuthoritiesConverter(xxxJwtGrantedAuthoritiesConverter));
	return http.build();
}
```

Reply:
@moarychan
Thank you for your answer, I am still not clear how do we check the expected authority to be checked using this 
`Converter<Jwt, Collection<GrantedAuthority>> xxxJwtGrantedAuthoritiesConverter = null`.

Previously I was checking,

`http.requestMatchers()
          .antMatchers("/api/v1/products")
          .authorizeRequests(requests -> requests.anyRequest().authenticated())
          .oauth2ResourceServer().jwt()
          .jwtAuthenticationConverter(new AADJwtBearerTokenAuthenticationConverter("roles", "ROLE_admin"))`

Reply:
@navin11 thanks for the update!

Please try to use the below `authenticationConverter ` instead of `AADJwtBearerTokenAuthenticationConverter`:

```java
JwtAuthenticationConverter authenticationConverter = new JwtAuthenticationConverter();
Converter<Jwt, Collection<GrantedAuthority>> jwtGrantedAuthoritiesConverter = new AadJwtGrantedAuthoritiesConverter();
Map<String, String> claimToAuthorityPrefixMap = new HashMap<>();
claimToAuthorityPrefixMap.put("roles", "ROLE_admin");
authenticationConverter.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthoritiesConverter);
```